transport/http2_client.go

```diff
@@ -232,6 +236,31 @@ func newHTTP2Client(ctx context.Context, addr TargetInfo, opts ConnectOptions) (
 	}
 	if opts.InitialWindowSize >= defaultWindowSize {
 		t.initialWindowSize = opts.InitialWindowSize
+		dynamicWindow = false
+	}
+	if dynamicWindow {
+		t.bdpEst = &bdpEstimator{
+			bdp: initialWindowSize,
+			updateFlowControl: func(n uint32) {
```

Comment:
Can you add a method to http2Client to do this instead of inlining it, please?  So this would be:

```go
    updateFlowControl: t.updateFlowControl
```